server/http_auth.go

```diff
@@ -0,0 +1,78 @@
+package server
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/evcc-io/evcc/core/auth"
+)
+
+var authCookieName = "auth"
+
+func setPasswordHandler(auth auth.API) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if err := auth.SetAdminPassword(r.FormValue("password")); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+	}
+}
+
+func loginHandler(auth auth.API) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		password := r.FormValue("password")
+		if !auth.IsAdminPasswordValid(password) {
+			http.Error(w, "Invalid password", http.StatusUnauthorized)
+			return
+		}
+
+		lifetime := time.Hour * 24 // 1 day valid
+		tokenString, err := auth.GenerateJwtToken(lifetime)
+		if err != nil {
+			http.Error(w, "Failed to generate JWT token: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		http.SetCookie(w, &http.Cookie{
+			Name:     authCookieName,
+			Value:    tokenString,
+			Path:     "/",
+			HttpOnly: true,
+			Expires:  time.Now().Add(lifetime),
+		})
+	}
+}
+
+func logoutHandler(w http.ResponseWriter, r *http.Request) {
+	http.SetCookie(w, &http.Cookie{
+		Name:     authCookieName,
+		Value:    "",
+		Path:     "/",
+		HttpOnly: true,
+		Expires:  time.Now().Add(-1),
```

Comment:
```suggestion
		MaxAge:  0,
```
-1 ms abzuziehen könnte möglicherweise bei Zeitunterschieden zwischen Client und Server zu unerwarteten Ergebnissen führen.